python/tvm/contrib/pipeline_executor.py

```diff
@@ -0,0 +1,533 @@
+        -------
+        mods : List[Module]
+            Module list.
+
+        mod_config : str
+            mods configuration
+        """
+
+        mods = []
+        for pipeline_mod in pipeline_mods:
+            mod = graph_executor.GraphModule(
+                pipeline_mod["default"](pipeline_mods[pipeline_mod]["dev"])
+            )
+            mods.append(mod.module)
+
+        return mods, json.dumps(mod_config)
+
+
+class PipelineConfig(object):
+    """The wrapper of each module to be pipelined. The wrapper mainly includes the
+    module itself as well as the binding that represents the connections of this
+    module's inputs and outputs to other modules.
+    """
+
+    class Binding:
+        """The class that use to storage module connection information.
+        The binding can be either "input" or "output".
+
+        Parameters
+        ----------
+        owner : ModuleWrapper
+            The class that own this interface, in such class there are
+            Module information like idx, module name
+
+        io_type : str
+            The type of this binding. It can be either "input" or "output".
+
+        name : str/integer
+            Binding name, for input it is string such as "data0";
+            for output it is the idx integer such as 0.
+        """
+
+        def __init__(self, owner, stype, name, data_type=None):
+            self.io_owner = owner
+            self.io_type = stype
+            self.name = str(name)
+            # These item that have dependency relation with self
+            self.bindings = []
+            # The item that self depend
+            self.parents = []
+
+            self.data_type = data_type
+
+        def get_name(self):
+            """Get owner name and self name"""
+            owner_name = ""
+            if isinstance(self.io_owner, PipelineConfig.ModuleWrapper):
+                owner_name = self.io_owner.name
+
+            return owner_name, self.name
+
+        def get_owner_idx(self):
+            """Return owner idex if owner is ModuleWrapper, if not return 0"""
+            if isinstance(self.io_owner, PipelineConfig.ModuleWrapper):
+                return self.io_owner.idx
+
+            # If not ModuleWrapper then owner is PipelineConfig, return 0
+            # to identify this is global interface
+            return 0
+
+        def is_global_interface(self):
+            """Check if this interface is global interface."""
+            return not isinstance(self.io_owner, PipelineConfig.ModuleWrapper)
+
+        def __repr__(self):
+            """Get all binding(input data), exepect like |data_0: mod1:data_0"""
+            ret = "  |{}: ".format(self.name)
+            for binding in self.bindings:
+                mname, dname = binding.get_name()
+                ret += "{0}:{1} ".format(mname, dname)
+            return ret
+
+        def check_dag_acyclic(self, start, inputs):
+            """Check if the DAG that current binding stay is acircle"""
```

Comment:
Fix the type and improve this sentence. What is meant by "current binding stay".

Reply:
changed into "Here to check if the DAG that build by bindings connections is acyclic."